File: demo/springboot/demo/src/main/java/com/demo/domain/dto/http/resonse/UserAuthResponse.java
package com.demo.domain.dto.http.resonse;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

@Data
public class UserAuthResponse {
    /** Wire as JSON string so JS clients keep full 64-bit precision; Java type stays Long. */
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private Long userId;
    private String account;
    private String name;
    private String avatarUrl;
    private String accessToken;
}

File: demo/flutter/flutternew/lib/domain/dto/auth/user_auth_response.dart
class UserAuthResponse {
  final int? userId;
  final String? account;
  final String? name;
  final String? avatarUrl;
  final String? accessToken;

  UserAuthResponse({
    required this.userId,
    required this.account,
    required this.name,
    required this.avatarUrl,
    required this.accessToken,
  });

  factory UserAuthResponse.fromJson(Map<String, dynamic> json) {
    return UserAuthResponse(
      userId: (json['userId'] as num?)?.toInt(),
      account: json['account'] as String?,
      name: json['name'] as String?,
      avatarUrl: json['avatarUrl'] as String?,
      accessToken: json['accessToken'] as String?,
    );
  }
}
